products/public-websites/cerner/postmortem/missing-content-schedule-appt-page.md

```diff
@@ -0,0 +1,103 @@
+
+## Root Causes
+
+_What caused this situation?_
+
+This situation was caused by 2 3-digit TTY 711 numbers that were using the `formation-react` `<Telephone />` component. These components required a prop called `pattern` that was not present, causing a runtime error, which resulted in the React widget not rendering.
+
+## Resolution
+_What steps were taken to resolve the issue?_ A PR that added the `pattern` prop to both `<Telephone />` components and an off-schedule deploy.
+
+_Date/time of resolution_ At 5:52pm EST on August 17, 2020.
+
+## Documentation links
+
+- [Original PR with bug](https://github.com/department-of-veterans-affairs/vets-website/pull/13882)
+- [PR with bug fix](https://github.com/department-of-veterans-affairs/vets-website/pull/13896)
+- [Slack discussion in #vfs-platform-support with request of off-schedule deploy](https://dsva.slack.com/archives/CBU0KDSB1/p1597695207118200)
+- [Slack discussion in #vsa-public-websites](https://dsva.slack.com/archives/C52CL1PKQ/p1597695037082300)
+
+## Action Items
+
+| Description                    | Type    | Owner        | Issue # |
+| ------------------------------ | ------- | ------------ | ------- |
+| Brainstorm alternatives of storing authenticated content in React widgets as opposed to Drupal | Frontend | Public Websites / CMS | Ticket not created yet |
+| Implement some form of smoke tests for website templates | Frontend | TBD | https://github.com/department-of-veterans-affairs/va.gov-team/issues/12445 |
+| Write unit tests that do not shallow render components for the Cerner page React widgets (the runtime error would have been caught this way). This may be a good use-case for React Testing Library. | Frontend | Public Websites | Ticket not created yet |
+| Fix `formation-react`'s `<Telephone />` component's `pattern` prop to inherently support 3-digit numbers without a prop needing to be passed in. | Frontend | TBD | Ticket not created yet |
+
+## Lessons Learned
+
+We need increased quality assurance, perhaps as a high-level form of smoke test, that runs against the CMS templates of the website, to ensure certain components and elements exist on key pages.
+
+### What went well
+
+Root cause was discovered and resolved quickly once the issue was noticed.
+
+### What went wrong
+
+There was an hour gap between the time the PR was merged and the deployment. This was due to needing an off-schedule deployment authorization after hours and not having immediate knowledge of backup authorizers when one authorizer isn't available after hours.
+
+### Where we got lucky
+
+Deployment resolving the issue when very smoothly and the involved engineers on VSA and VSP were all very supportive.
+
+### Takeaways
+
+We have long identified the content templates of the FE code to need better test coverage, and this issue demonstrates the increasing importance as we continue to scale.
```

Comment:
This is sort of outside of templates in that it's a React widget but it is similar. We would typically perform tests at a high-level via e2e tests but couldn't because it had the Drupal page as a dependency. I think a collection of unit tests that do full, deep renders of the React component trees across our FE widgets would be a useful precedent to set.